### PR TITLE
[Fix] `DayPicker{Range,SingleDate}Controller`: Fix incorrect firstDayOfWeek null checking

### PR DESCRIPTION
**Pull request by axsann**
Fix incorrect firstDayOfWeek null checking in `isFirstDayOfWeek` and `isLastDayOfWeek`.

`||` operator will use right side value when 0 was given.
Even if firstDayOfWeek set to Sunday (0), the method will use default moment value.

### CODE REVIEW

**Comment by ljharb** [comment]
@axsann unfortunately, since you made this PR from a fork you don't own, i can't force push to it and thus can't land the PR. Can you add me to https://github.com/wamazing/react-dates, so I can get this PR landed?

**Comment by ljharb** [comment]
cc @matis-schotte; it'd really help me out if you could temporarily add me to https://github.com/wamazing/react-dates so i can get this PR landed 🙏

**Comment by axsann** [comment]
@ljharb 
Sorry,  I am no longer under employment at wamazing.
so I have no permission.

What would you suggest I do?

**Comment by ljharb** [comment]
@axsann thanks for replying! no worries, let's keep this open in the hopes that someone at wamazing can still help resolve it.

If not, I can cherry-pick directly (so no need for a new PR), but I'd very much prefer to actually update this PR.

**Comment by axsann** [comment]
@ljharb 
Thank you.
I will try to contact my former colleague!
Please wait a little bit.

**Comment by ljharb** [comment]
Thanks @whirakawa! Once this PR is merged, feel free to remove my access.

**Review by ljharb — APPROVED** [review]

**Comment by ljharb** [comment]
Thanks! Sorry for the delay :-)

**Comment by dotfury** [comment]
Hi @ljharb ,
I am also an employee at WAmazing and wanted to clarify something. Since this has been merged now, do need to wait for a new release of `react-dates`? Or can we just remove the patch without a `react-dates` update?

**Comment by ljharb** [comment]
@dotfury yes, you do need to wait for a release, but I'm hoping to publish one this week.

**Pull request merged**
